Make all suggested Deposit changes

4855062503 opened the pull request:
- Close button Says 'Close' for deposit page after creating LN deposit request, else 'Cancel' (for Bitcoin deposit requests)
- Make Withdrawal buttons all Bronze gradient
- OK button bronze gradient
- ListTile text all center aligned
- Widgets in Column spaced accordingly
- Copy to Clipboard button added, in prototyped colour
- Uses ColumnView instead of formWidgetsWidth in SizedBox

djpnewton commented:
1. `BronzeRoundedButton` holePunch variable does not do anything? if not best to remove it
2. Various places have changed the `fillGradient` parameter of calls to `BronzeRoundedButton`, if so you should also change the textColor and fillColor parameters to match 
3. might as well change the 'OK' from the LN deposit confirmed button to 'Close' so it matches the other close buttons

also can you post some screenshots

4855062503 commented:
> 1. `BronzeRoundedButton` holePunch variable does not do anything?

`holePunch` adds the arrow in our Figma  "Continue" button design.

I have added the rest of the suggestions.

djpnewton commented:
> > 1. `BronzeRoundedButton` holePunch variable does not do anything?
> 
> `holePunch` adds the arrow in our Figma "Continue" button design.

Ok maybe better to rename it to something like 'arrowIcon' or something

> I have added the rest of the suggestions.

You missed some instances of changing the textColor and fillColor parameters. Can you please do more self-review of your work before thinking it is done

4855062503 commented:
@djpnewton 

I changed the fill params so that if I replaced `ZapPrimaryGradient` or `ZapSecondaryGradient`, I replaced the other params with a Color class. 

Replaced `holePunch` occurrences with `fwdArrow`.

djpnewton commented:
this is the exchange input:
![image](https://user-images.githubusercontent.com/42459/174708188-2479fc8f-d422-4274-b14f-1c02246c0486.png)

here is the LN deposit input:
![image](https://user-images.githubusercontent.com/42459/174708356-b937edf0-624c-400b-80be-9e59fd0e5733.png)
![image](https://user-images.githubusercontent.com/42459/174708399-0968ce6b-5f2e-482c-95b8-01103d587ab9.png)

1. the input should be the same width as the button
2. make it look like the exchange input (put the units to the right, keep the outline) 
3. make a class like `AmountInput` and put it in widgets.dart then we can more easily keep all these inputs consistent by using the same class

djpnewton commented on lib/deposit.dart:
```diff
@@ -318,7 +321,7 @@ class _CryptoDepositDetailScreenState extends State<CryptoDepositDetailScreen> {
                   child: QrImage(
-            backgroundColor: ZapSurface,
+            backgroundColor: Colors.transparent,
             foregroundColor: ZapOnSurface,
```
```suggestion
            foregroundColor: Colors.white,
```

djpnewton commented on lib/deposit.dart:
```diff
@@ -387,18 +394,19 @@ class _CryptoDepositNewScreenState extends State<CryptoDepositNewScreen> {
                           child: QrImage(
-                        backgroundColor: ZapSurface,
+                        backgroundColor: Colors.transparent,
                         foregroundColor: ZapOnSurface,
```
```suggestion
                        foregroundColor: Colors.white,
```

djpnewton commented on lib/deposit.dart:
```diff
@@ -387,18 +394,19 @@ class _CryptoDepositNewScreenState extends State<CryptoDepositNewScreen> {
-                          onPressed: _copyRecipient, icon: Icon(Icons.copy)))
+                  BronzeRoundedButton(_copyRecipient, Colors.white,
+                      Color(0xff32333b), null, 'Copy to Clipboard',
+                      width: 320, height: 65, icon: Icons.copy),
```
we have `ButtonWidth` and ButtonHeight constants

djpnewton commented on lib/deposit.dart:
```diff
@@ -387,18 +394,19 @@ class _CryptoDepositNewScreenState extends State<CryptoDepositNewScreen> {
                       padding: EdgeInsets.all(10)),
-                  ListTile(
-                      leading: SizedBox(),
-                      title: Center(child: Text(shortenStr(widget.recipient))),
```
show the recipient under the QR code

djpnewton commented on lib/deposit.dart:
```diff
@@ -613,7 +624,7 @@ class _FiatDepositDetailScreenState extends State<FiatDepositDetailScreen> {
                     QrImage(
-                      backgroundColor: ZapSurface,
+                      backgroundColor: Colors.transparent,
                       foregroundColor: ZapOnSurface,
```
```suggestion
                      foregroundColor: Colors.white,
```

djpnewton commented on lib/exchange.dart:
```diff
@@ -386,18 +380,13 @@ class _ExchangeWidgetState extends State<ExchangeWidget> {
-              color: ZapSurface,
-              child: TextField(
-                  controller: _receiveController,
-                  readOnly: true,
```
you need a `readOnly` field for `BronzeValueInput`

djpnewton replied:
still a problem

djpnewton commented on lib/widgets.dart:
```diff
@@ -5,6 +5,39 @@ import 'package:zapdart/colors.dart';
+        dottedBorder: true,
+        color: ZapSurface,
+        child: TextFormField(
+          controller: this.controller == null ? null : this.controller,
```
```suggestion
          controller: this.controller,
```

djpnewton commented on lib/widgets.dart:
```diff
@@ -5,6 +5,39 @@ import 'package:zapdart/colors.dart';
+        child: TextFormField(
+          controller: this.controller == null ? null : this.controller,
+          decoration: InputDecoration(
+              suffixText: this.suffixText == null ? null : this.suffixText,
```
```suggestion
              suffixText: this.suffixText,
```

djpnewton reviewed (changes requested):
make the deposit input fields the same width as the buttons (might have to fudge it a bit for the button rounded edges)

djpnewton commented on lib/event.dart:
```diff
@@ -124,33 +124,34 @@ class _DepositAmountScreenState extends State<DepositAmountScreen> {
                       child: Form(
                           key: _formKey,
                           child: Container(
+                              width: ButtonWidth + 14.0,
```
put a comment why we have the girth fudge factor here

4855062503 commented:
@djpnewton 
Have added fudge factor comment + readOnly property to exchange widget on receive end

djpnewton commented on lib/widgets.dart:
```diff
@@ -5,6 +5,43 @@ import 'package:zapdart/colors.dart';
+        child: TextFormField(
+          controller: this.controller,
+          readOnly:
+              (this.readOnly == true && this.readOnly != null) ? true : false,
```
```suggestion
              this.readOnly ?? false,
```

djpnewton commented:
the bank deposit page should probably have all the fields the `ButtonWidth` width and have a cancel button to make it look more like the other deposit options:

![image](https://user-images.githubusercontent.com/42459/175860590-0111eba3-3de5-4924-ae5e-a797aaa90df0.png)

djpnewton commented:
the CI check is failing too

djpnewton commented on lib/config.dart:
```diff
@@ -17,10 +17,10 @@ const ButtonHeight = 65.0;
 
 // the default testnet value
-const _DefaultTestnet = true;
+const _DefaultTestnet = false;
```
accidental change

djpnewton commented on lib/config.dart:
```diff
@@ -17,10 +17,10 @@ const ButtonHeight = 65.0;
 
 // beryllium settings
-const String _BeServerUrl = 'https://beryllium-test.zap.me/';
+const String _BeServerUrl = 'https://beryllium.zap.me/';
```
accidental change

djpnewton reviewed (changes requested):
should probably have the big down arrow icon on the bank deposit page too 
![image](https://user-images.githubusercontent.com/42459/175893959-af9565b8-90f5-410c-af00-ad30fcdcbf55.png)
